augur/filter.py

```diff
@@ -146,7 +188,8 @@ def run(args):
     #If VCF, open and get sequence names
     if is_vcf:
         seq_keep, all_seq = read_vcf(args.sequences)
-    else:
+        # TODO: How to best handle "source of truth" for VCF files vs. metadata?
+    elif args.sequences or args.sequence_index:
```

Comment:
One issue I just ran into while mocking up a ncov workflow with this interface was that I had to include a sequence index in the final filter step that pulls in all of the subsampled strains and extracts their FASTA sequences. Even though the index isn't necessary at that stage because there are no sequence-based filters being applied, if I didn't include the sequence index, augur would try to build it for me.

So, this line here should probably also check for whether any sequence-based filters have been requested (as in the argument validation code above) and then skip the sequence index completely if it isn't needed.